projects/developer/src/app/common/services/color.service.ts

```diff
@@ -21,6 +21,8 @@ export class ColorService {
     public static SCALE_BLUE3 = '#24567F';
     public static RECIPE_NODE = '#777';
     public static WARNING = '#fdb813';
+    public static FONT_DARK_THEME = '#d9d9d9';

```

Comment:
I'm wondering if these could be moved into the themes directly instead of the color service? I don't really understand what the color service is for tbh, but could we use theme service instead?

```ts
// class variable, where chart options are defined
// initialize the colors to null or empty strings?
public chartOptions = {
  ...
  whateverNestedThings: {
    fontColor: null
  }
}

// ngOnInit, use the existing text color defined in the theme
ngOnInit() {
  const updateChartColor = () => {
    const color = this.themeService.getProperty('--main-text');
    this.chartOptions.whateverNestedThing.font = color;
  };

  // don't even need to care about the theme name
  this.themeService.themeChange.subscribe(() => updateChartColor());
}
```

Reply:
I was using color service as I saw it being used already for other parts of the components, although using `this.themeService.getProperty()` makes more sense looking back at it.